sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicyOptions.java

```diff
@@ -0,0 +1,101 @@
+public class RetryPolicyOptions {
+
+    private final RetryStrategy retryStrategy;
+    private final String retryAfterHeader;
+    private final ChronoUnit retryAfterTimeUnit;
+
+    /**
+     * Creates a default {@link RetryPolicyOptions} used by a {@link RetryPolicy}. This will use
+     * {@link ExponentialBackoff} as the {@link #getRetryStrategy retry strategy} and will ignore retry delay headers.
+     */
+    public RetryPolicyOptions() {
+        this(new ExponentialBackoff(), null, null);
+    }
+
+    /**
+     * Creates the {@link RetryPolicyOptions} with provided {@link RetryStrategy} that will be used when a request is
+     * retried. It will ignore retry delay headers.
+     *
+     * @param retryStrategy The {@link RetryStrategy} used for retries.
+     * @throws NullPointerException if {@code retryStrategy} is {@code null}.
+     */
+    public RetryPolicyOptions(RetryStrategy retryStrategy) {
+        this(retryStrategy, null, null);
+    }
+
+    /**
+     * Creates the {@link RetryPolicyOptions} with provided {@code retryAfterHeader} and {@code retryAfterTimeUnit}
+     * that will be used when a request is retried. This will use {@link ExponentialBackoff} as the
+     *  {@link #getRetryStrategy retry strategy}
+     *
+     * @param retryAfterHeader The HTTP header, such as 'Retry-After' or 'x-ms-retry-after-ms', to lookup for the
+     * retry delay. The value {@code null} is valid.
+     * @param retryAfterTimeUnit The time unit to use when applying the retry delay. {@code null} is valid if, and only
+     * if, {@code retryAfterHeader} is {@code null}.
+     * @throws NullPointerException Only if {@code retryAfterTimeUnit} is {@code null} and {@code retryAfterHeader}
+     * is not {@code null}.
+     */
+    public RetryPolicyOptions(String retryAfterHeader, ChronoUnit retryAfterTimeUnit) {
+        this(new ExponentialBackoff(), retryAfterHeader, retryAfterTimeUnit);
+    }
+
+    /**
+     * Creates the {@link RetryPolicyOptions} with provided {@link RetryStrategy}, {@code retryAfterHeader} and
+     * {@code retryAfterTimeUnit} that will be used when a request is retried.
+     *
+     * @param retryStrategy The {@link RetryStrategy} used for retries.
+     * @param retryAfterHeader The HTTP header, such as 'Retry-After' or 'x-ms-retry-after-ms', to lookup for the
+     * retry delay.The value {@code null} is valid.
+     * @param retryAfterTimeUnit The time unit to use when applying the retry delay. {@code null} is valid if, and only
+     *      * if, {@code retryAfterHeader} is {@code null}.
+     * @throws NullPointerException if {@code retryStrategy} is {@code null}. Also when {@code retryAfterTimeUnit}
+     * is {@code null} and {@code retryAfterHeader} is not {@code null}.
+     */
+    public RetryPolicyOptions(RetryStrategy retryStrategy, String retryAfterHeader, ChronoUnit retryAfterTimeUnit) {
+        this.retryStrategy = Objects.requireNonNull(retryStrategy, "'retryStrategy' cannot be null.");
```

Comment:
Should this be using `requireNonNull` or just use `ExponentialBackoff` if it is null?

Reply:
will use use ExponentialBackoff